cli/azd/pkg/commands/pipeline/github_provider.go

```diff
@@ -305,10 +307,33 @@ func (p *GitHubCiProvider) configureConnection(
 	console.Message(ctx, "Setting AZURE_CREDENTIALS GitHub repo secret.\n")
 
 	ghCli := github.NewGitHubCli(ctx)
+	// set azure credential for pipelines can log in to Azure
 	if err := ghCli.SetSecret(ctx, repoSlug, "AZURE_CREDENTIALS", string(credentials)); err != nil {
 		return fmt.Errorf("failed setting AZURE_CREDENTIALS secret: %w", err)
 	}
 
+	if infraOptions.Provider == "terraform" {
+		// terraform expect the credential info to be set in the env individually
+		type credentialParse struct {
+			Tenant       string `json:"tenant"`
+			ClientId     string `json:"appId"`
+			ClientSecret string `json:"password"`
+		}
+		values := credentialParse{}
+		if e := json.Unmarshal(credentials, &values); e != nil {
+			return fmt.Errorf("setting terraform env var credentials: %w", e)
+		}
+		if err := ghCli.SetSecret(ctx, repoSlug, "ARM_TENANT_ID", values.ClientId); err != nil {
+			return fmt.Errorf("setting terraform env var credentials:: %w", err)
+		}
+		if err := ghCli.SetSecret(ctx, repoSlug, "ARM_CLIENT_ID", values.ClientId); err != nil {
+			return fmt.Errorf("setting terraform env var credentials:: %w", err)
+		}
+		if err := ghCli.SetSecret(ctx, repoSlug, "ARM_CLIENT_SECRET", values.ClientId); err != nil {
+			return fmt.Errorf("setting terraform env var credentials:: %w", err)
+		}
+	}
```

Comment:
This is a good workaround for now.  We should think about a longer term fix where the we can keep the interaction more loosely coupled.